dpgen/generator/lib/model_devi_calypso.py

```diff
@@ -0,0 +1,452 @@
+"""
+calypso model devi:
+       1. gen_structures
+       2. analysis
+       3. model devi
+"""
+
+import copy
+import dpdata
+import math
+import numpy as np
+import os
+import random
+import re
+import glob
+import shutil
+from ase.io.vasp import write_vasp
+from ase.io.trajectory import Trajectory
+#from deepmd.infer import calc_model_devi
+#from deepmd.infer import DeepPot as DP
+from pathlib import Path
+from distutils.version import LooseVersion
+from dpgen import dlog
+from dpgen.generator.lib.utils import make_iter_name
+from dpgen.generator.lib.calypso import write_model_devi_out
+from dpgen.generator.lib.parse_calypso import _parse_calypso_input,_parse_calypso_dis_mtx
+from dpgen.dispatcher.Dispatcher import Dispatcher, _split_tasks, make_dispatcher, make_submission
+
+train_name = '00.train'
+model_devi_name = '01.model_devi'
+fp_name = '02.fp'
+calypso_run_opt_name = 'gen_stru_analy'
+calypso_model_devi_name = 'model_devi_results'
+
+def gen_structures(iter_index,jdata,mdata):
+
+    # run calypso
+    # vsc means generate elemental, binary and ternary at the same time
+    vsc = jdata.get('vsc',False)  # take CALYPSO as confs generator
+
+    model_devi_group_size = mdata['model_devi_group_size']
+    model_devi_resources = mdata['model_devi_resources']
+
+
+    iter_name = make_iter_name(iter_index)
+    work_path = os.path.join(iter_name, model_devi_name)
+    assert(os.path.isdir(work_path))
+    # 
+    calypso_run_opt_path = os.path.join(work_path,calypso_run_opt_name)
+    calypso_model_devi_path = os.path.join(work_path,calypso_model_devi_name)
+    #
+
+    calypso_path = jdata.get('calypso_path')
```

Comment:
The parameters about machines and environment variables should be added to `machine.json`, not `param.json`